docs/operations/single-server.md

```diff
@@ -42,37 +43,43 @@ The startup scripts for these example configurations run a single ZK instance al
 
 The example configurations run the Druid Coordinator and Overlord together in a single process using the optional configuration `druid.coordinator.asOverlord.enabled=true`, described in the [Coordinator configuration documentation](../configuration/index.md#coordinator-operation).
 
+The `start-druid` is a generic launch script for starting druid services on single server, it accepts optional arguments like services, memory and config.
+All reference configurations can be acheived by passing appropriate arguments to this script.
+Existing launch scripts are deprecated and will be removed in the next release. 
+
+
 While example configurations are provided for very large single machines, at higher scales we recommend running Druid in a [clustered deployment](../tutorials/cluster.md), for fault-tolerance and reduced resource contention.
 
 ## Single server reference configurations
 
-### Nano-Quickstart: 1 CPU, 4GiB RAM
+### Nano: 1 CPU, 4GiB RAM
 
-- Launch command: `bin/start-nano-quickstart`
-- Configuration directory: `conf/druid/single-server/nano-quickstart`
+- Launch command: `bin/start-druid --memory=4g`
+- Configuration directory: `conf/druid/single-server/quickstart`
 
-### Micro-Quickstart: 4 CPU, 16GiB RAM
+### Micro: 4 CPU, 16GiB RAM
 
-- Launch command: `bin/start-micro-quickstart`
-- Configuration directory: `conf/druid/single-server/micro-quickstart`
+- Launch command: `bin/start-druid --memory=16g`
+- Configuration directory: `conf/druid/single-server/quickstart`
 
 ### Small: 8 CPU, 64GiB RAM (~i3.2xlarge)
 
-- Launch command: `bin/start-small`
-- Configuration directory: `conf/druid/single-server/small`
+- Launch command: `bin/start-druid --memory=64g`
+- Configuration directory: `conf/druid/single-server/quickstart`
 
 ### Medium: 16 CPU, 128GiB RAM (~i3.4xlarge)
 
-- Launch command: `bin/start-medium`
-- Configuration directory: `conf/druid/single-server/medium`
+- Launch command: `bin/start-druid --memory=128g`
+- Configuration directory: `conf/druid/single-server/quickstart`
 
 ### Large: 32 CPU, 256GiB RAM (~i3.8xlarge)
 
-- Launch command: `bin/start-large`
-- Configuration directory: `conf/druid/single-server/large`
+- Launch command: `bin/start-druid --memory=256g`
+- Configuration directory: `conf/druid/single-server/quickstart`
 
 ### X-Large: 64 CPU, 512GiB RAM (~i3.16xlarge)
 
-- Launch command: `bin/start-xlarge`
-- Configuration directory: `conf/druid/single-server/xlarge`
+- Launch command: `bin/start-druid --memory=512g`
+- Configuration directory: `conf/druid/single-server/quickstart`
 
+Memory argument (`--memory`) in the above launch command is optional, if not specified Druid will use upto 80% of system memory.
```

Comment:
I feel like this is a really key point! It is really buried here. I missed it after reading the docs twice and discovered this by playing and running `bin/start-druid --verbose` do we even need to list all the configurations above?

Reply:
I think we need a separate section that explains the usage of `start-druid` and the calls out the key points. It can mostly be taken from the description of this PR. @findingrish , maybe you could add a header right before `Single server reference configurations` and briefly explain the usage and key points such as this point about the memory.

Reply:
I will try to place it right where we introduce the `start-druid` script. 

> do we even need to list all the configurations above?

I feel it might help the users choose a machine for trying things out.

Reply:
> I feel it might help the users choose a machine for trying things out.

This is a good point! It is helpful to anchor people on a few specific sizes in case they are picking a VM rather than their laptop.